pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -162,6 +162,20 @@ func buildLogsTimeSeriesFilterQuery(fs *v3.FilterSet, fields map[string]v3.Attri
 			}
 		}
 	}
+
+	// add group by conditions to filter out log lines which doesn't have the key
+	for _, attr := range groupBy {
+		enrichedAttr, err := encrichFieldWithMetadata(attr, fields)
+		if err != nil {
+			return "", err
+		}
+		if !enrichedAttr.IsColumn {
+			columnType := getClickhouseLogsColumnType(enrichedAttr.Type)
+			columnDataType := getClickhouseLogsColumnDataType(enrichedAttr.DataType)
+			conditions = append(conditions, fmt.Sprintf("indexOf(%s_%s_key, '%s') > 0", columnType, columnDataType, enrichedAttr.Key))
```

Comment:
Is there any difference b/w `has` vs `IndexOf`?

Reply:
Same question @nityanandagohain 
Is there any perf difference?